frontend/src/components/Playback/SubtitleTrack.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div
+    class="uno-absolute uno-bottom-0 uno-left-0 uno-w-full uno-text-center"
+    :label="playerElement.currentExternalSubtitleTrack?.label"
+    :srclang="playerElement.currentExternalSubtitleTrack?.srcLang"
+    :src="playerElement.currentExternalSubtitleTrack?.src">
+    <span
+      class="uno-inline-block"
+      :style="subtitleStyle">
+      <JSafeHtml
+        v-if="currentSubtitle !== undefined"
+        :html="currentSubtitle.text" />
+      {{ previewText }}
+    </span>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+import { clientSettings } from '@/store/client-settings';
+import { mediaControls } from '@/store';
+import { playerElement } from '@/store/player-element';
+import { isNil } from '@/utils/validation';
+import type { ParsedSubtitleTrack } from '@/utils/subtitles';
+
+defineProps<{
+  previewText?: string;
+}>();
+
+/**
+ * Function to find the current subtitle based on the current time.
+ * It starts searching from the last found index to optimize the search.
+ * If not found, it searches from the beginning of the list.
+ */
+let lastIndex = 0; // Variable to store the last found index of the subtitle
+const findCurrentSubtitle = (data: ParsedSubtitleTrack, currentTime: number) => {
+  // Start searching from the last found index
+  for (let i = lastIndex; i < data.dialogue.length; i++) {
+    const subtitle = data.dialogue[i];
+
+    if (subtitle.start < currentTime && subtitle.end > currentTime) {
+      lastIndex = i; // Update the last found index
+
+      return subtitle;
+    } else if (subtitle.start > currentTime) {
+      break;
+    }
+  }
+
+  // Start searching from the beginning
+  for (const [i, subtitle] of data.dialogue.entries()) {
+    if (subtitle.start < currentTime && subtitle.end > currentTime) {
+      lastIndex = i; // Update the last found index
+
+      return subtitle;
+    } else if (subtitle.start > currentTime) {
+      break;
+    }
+  }
+};
```

Comment:
I believe a more performant approach would be to have a Map (computed on the ``parseSsaFile`` of course):

* key is the timestamp rounded to only one decimal (I believe this should cover all cases keeping a good precision?)
* Value is the content of the subtitle

The computed property just needs to also do a ``Math.round`` to the current time and return the value of the map, if it's undefined, the [previous one](https://github.com/jellyfin/jellyfin-vue/blob/master/frontend/src/components/lib/JVirtual/JVirtual.vue#L200) is returned.

Reply:
I don't quite understand what you mean by previous one. If you're referring to the previous subtitle, I believe it would be disregarding the expected behavior of subtitles, since there can be intentional gaps between dialogue (Usually when there's no speech). Other than that I agree with refactoring this code to use a map since it would probably be more performant and easier to read.